### DISCUSSION TITLE
.NET 8 support?

### DISCUSSION THREAD

**Discussion by cactusoft** [Blazored/Modal#537] [General]
First off, have used this package in .NET6 for a year or so and have found it simple to use and reliable. Great work!

I am in the process of upgrading my .NET6 Blazor project to .NET 8 (rc2) and have found quite a few more breaking changes than I expected. Even in areas I didn't think would break. One of those appears to be Blazored.Modal. Have set things up in the same way, using latest package, but it just gives a null instance error when I try to launch a popup.

I have tried the CascadingBlazoredModal tags in both the App and Routes .razor components as the wiring in .NET is a little different, but doesn't seem to matter. There are no build errors, the error only occurs at run time when a model is triggered.

I would not rule out the possibility I am doing something wrong, just wasn't sure if anyone has got the existing package working with Blazor server .NET 8 (I have stuck to just the server version to try to keep things as similar as possible to the existing .NET project), or if there are changes required for .NET 8?

**Comment by cactusoft** [comment]
I think I figured out the issue, and a fix.

In the .NET 8 default project, pages are static unless explicitly set to be InteractiveServer (essentially like the "Server" mode in .NET 6). Even if you set rendering explicitly in individual files, the MainLayout etc won't have that interactive functionality. You can set the old style .NET 6 "always on" connection behaviour globally from the routes tag in the App.razor.

<Routes @rendermode="RenderMode.InteractiveServer" />

Also, do the same to the HeadOutlet in the App.razor:

<HeadOutlet @rendermode="RenderMode.InteractiveServer" />

After this, the Blazored.Modal dialogs work as expected.

One thing to note however is that this breaks the Account pages in the identity system, they will generate errors. But the issue is noted on the aspnetcore Github bug list and supposedly this will be fixed with the GA release. I think they will force the Account pages to run statically, regardless of the Routes tag. So for now, I am setting that rendermode to interactive so I can get all my old .NET 6 functionality into .NET 8 and will wait for GA to fix the account pages.

**Reply by schuchuva** [reply]
Not working for Blazor 8 app in InteractiveWebAssembly mode. Created brand new project with global InteractiveWebAssembly mode and got runtime error: Unhandled exception rendering component: Cannot provide a value for property 'ModalService' on type 'Blazored.Modal.CascadingBlazoredModal'. There is no registered service of type 'Blazored.Modal.Services.IModalService'.
System.InvalidOperationException: Cannot provide a value for property 'ModalService' on type 'Blazored.Modal.CascadingBlazoredModal'. There is no registered service of type 'Blazored.Modal.Services.IModalService'.

**Reply by chrissainty** [reply]
Sounds like you haven't registered the services in the `Program.cs`. Have you added the following line:

```csharp
builder.Services.AddBlazoredModal();
```

**Reply by schuchuva** [reply]
Yes

**Reply by chrissainty** [reply]
I'll need a bit more to go on then a repro would be best.

**Reply by wallyhall** [reply]
@cactusoft - thank-you.  I wasn't sure what I was doing wrong - I don't think the docs are up to date for .NET 8 users.  (I'm new to both Blazor and .NET8... thought I was going mad!)

For other's reference, I've put a `<CascadingBlazoredModal />` tag in the specific InteractiveServer pages which I wish to use a Blazored Modal in - and that appears to work (without forcing the "old style .NET 6 always on behaviour").

**Reply by ZenBoer** [reply]
@wallyhall how do you reference the Model instance? I've added the tag <CascadingBlazoredModal /> but my Modal property stays null: [CascadingParameter] public IModalService Modal { get; set; } = default!;

**Reply by wallyhall** [reply]
@ZenBoer - I have to confess - subsequent to posting success, I had other weird bugs cropping up (with Blazored).  Eventually I just bit the bullet - and fully refactored to using [Blazor Bootstrap](https://docs.blazorbootstrap.com/).

With no disrespect felt or intended toward Blazored ... (I maintain a few tiny public GitHub repos myself, and it can be a serious demand on your spare time), I have found that Blazor Bootstrap is feature complete, stable, and well documented.

For popups/modals, [see here](https://docs.blazorbootstrap.com/components/modal#examples).

**Reply by 3hxx** [reply]
> I'll need a bit more to go on then a repro would be best.

@chrissainty Please update the docs to this, .NET 8+ has fucked it if you follow the docs. It's such a good library but the current implementation following docs does not work at all?
